### ISSUE TITLE
NVDA unable to navigate and focus an hyperlink if the paragraph is too long

### ISSUE THREAD

**Issue by ldominic-hy** [nvaccess/nvda#8180]
### Steps to reproduce:
> A list of the steps you take to demonstrate the problem. 
>
>Create an html file with the following:
>
    <head></head>
    <body>
        <p aria-hidden="false">
            This is a testing of the paragraph overflow.  The hyper link cannot be focused via the NVDA screen reader if this paragraph is too long.  If the paragraph is short enough, the screen reader can navigate and set focus.  Pressing the space bar will open the google.ca page.
            <a aria-label="Open google.ca" href="http://google.ca" target="_blank">
                <span>
                    <span>Open Google Not Working</span>
                </span>
            </a>
        </p>

        <p aria-hidden="false">
            This is a working version.  The hyper link can be focused via the NVDA screen reader if the line is broke up properly.
            <a aria-label="Open google.ca working" href="http://google.ca" target="_blank">
                <span>
                    <span>Open Google Working</span>
                </span>
            </a>
        </p>

        <p>More wordings </p>
    </body>
>
>
>
> 1. Open the above html using Chrome
> 2. Open NVDA and speech viewer
> 3. Navigate through the page using the arrow keys.

### Expected behavior:
> Both hyperlinks can be focus by NVDA and open another window to google.ca when spacebar is pressed.

### Actual behavior:
> The first hyperlink does not work and only the second hyperlink works.

### System configuration:
NVDA version:
> 2016.3

NVDA Installed or portable:
> NVDA installed

Other information:
> Running in a laptop

Windows version:
> Windows 7 or Windows 10

Name and version of other software in use when reproducing the issue:
> NA

### Other questions:

Does the issue still occur after restarting your PC?
> Yes

Have you tried any other versions of NVDA?
> No

**Comment by ldominic-hy** [comment]
Hi, thank you for your quick reply.  Just to further clarify, navigate through the page via tab is working fine.  The issue is about navigate through the page using the up and down arrows keys.  

The way I did this was by the following:

1. Navigate the paragraph using the arrow keys and use the speech viewer to view the wordings.
2. The screen reading would stop in the middle of the first paragraph as there is a limited number of characters that NVDA would read.
3. Hit the arrow key to continue the reading of the rest of the paragraph.

I notice for (3), the link and the last part of the paragraph goes together and the NVDA is unable to focus the link.  The second paragraph I made it short enough that there is no break of reading the whole paragraph and the hyperlink can be focus accordingly.

Hopefully this clarifies,

Thanks!

**Comment by derekriemer** [comment]
This works for me just fine. You simply have to arrow to the links, or use tab, or move by link (if in screen layout).

**Comment by ehollig** [comment]
As stated in https://github.com/nvaccess/nvda/issues/8180#issuecomment-381360473, this is expected behavior when screen layout is turned on. When screen layout is on, NVDA will announce buttons, links, edit fields, etc. on the same line if they appear that way visually as well. it sounds as if you would like the link to focus when you press the down arrow key. This can be accomplished by turning screen layout off with NVDA+V or insert+V in a web document or uncheck the Use screen layout (when supported) option in the Browse mode settings.

**Comment by feerrenrut** [comment]
I was unable to find any difference in behaviour between the two paragraphs. @ldominic-hy If I understand correctly, you would like the link being read to have focus, so that it can be activated. One difficulty is what should happen when there are several links within this paragraph? Perhaps each should have focus as it is read, and if you are quick enough you could activate it at that moment.

Alternatively, we have considered having screen layout disabled by default, meaning that you would need to press down arrow for each link, which can get annoying when there are many links.

**Comment by ldominic-hy** [comment]
Hi @ehollig, @feerrenrut and @derekriemer,

Thank you for your feedback and suggestion.  I turned off the screen layout option in the Browse Mode setting and it is working fine now.

Best regards,

**Comment by 707amitsingh** [comment]
@ehollig  @feerrenrut  @derekriemer  @Idominic-hy

I am facing the same issue and turning screen layout off does work as expected, but is there any other way to focus and read all the links in a long paragraph when a user is using UP/DOWN key to navigate through a page?
There is a possibility that an end user may not know that he has to turn-off screen layout mode and will skip some links.

Regards,
